networking_generic_switch/devices/netmiko_devices/__init__.py

```diff
@@ -58,23 +59,39 @@ def _format_commands(self, commands, **kwargs):
                                                       args=kwargs)
         return cmd_set
 
-    def send_commands_to_device(self, cmd_set):
+    def send_commands_to_device(self, cmd_set, num_retries=3, wait_on_fail=15):
         if not cmd_set:
             LOG.debug("Nothing to execute")
             return
-
+        for count in range(num_retries):
+            final_attempt = True if count == num_retries-1 else False
+            if self.send_commands_attempt(cmd_set, final_attempt):
+                return
+            err_msg = "Sending commands to device failed (attempt {})."
+            err_msg += "Waiting for {} seconds before retry."
+            LOG.error(err_msg.format(str(count+1), wait_on_fail))
+            time.sleep(wait_on_fail)
+
+    def send_commands_attempt(self, cmd_set, final_attempt=False):
         try:
             with netmiko.ConnectHandler(**self.config) as net_connect:
                 net_connect.enable()
-                output = net_connect.send_config_set(config_commands=cmd_set)
+                output = net_connect.send_config_set(
+                    config_mode_cmd='configure private', 
+                    config_commands=cmd_set)
                 # NOTE (vsaienko) always save configuration when configuration
                 # is applied successfully.
                 if self.SAVE_CONFIGURATION:
                     net_connect.send_command(self.SAVE_CONFIGURATION)
+            LOG.debug(output)
+            return True
         except Exception as e:
-            raise exc.GenericSwitchNetmikoConnectError(config=self.config,
-                                                       error=e)
-        LOG.debug(output)
+            if not final_attempt:
+                LOG.error(e)
```

Comment:
I think technically this this wants to be a warning log, would be an error log on the final error I think.